tests/posix/headers/src/stropts_h.c

```diff
@@ -0,0 +1,16 @@
+#include <zephyr/posix/stropts.h>
+#include "_common.h"
+/**
+ * @brief Test existence and basic functionality of stropts.h
+ *
+ * @see stropts.h
+ */
+ZTEST(posix_headers, test_stropts_h)
+{
+	zassert_not_null((void *)putmsg, "putmsg is null");
```

Comment:
As in other tests, if we are not linking POSIX objects, then we expect that there is no symbol that can be resolved for `putmsg`. However, users should still be able to include the header (and likely see some relevant bits).

```
/opt/toolchains/zephyr-sdk-0.16.4/arc-zephyr-elf/bin/../lib/gcc/arc-zephyr-elf/12.2.0/../../../../arc-zephyr-elf/bin/ld.bfd: /__w/zephyr/zephyr/tests/posix/headers/src/stropts_h.c:10: undefined reference to `putmsg'
```

So it only makes sense to assert that the symbol exists when we build that source.

```suggestion
#ifdef CONFIG_POSIX_API
	zassert_not_null((void *)putmsg, "putmsg is null");
#endif
```

Reply:
Should I also be doing conditional import as in
```c
#ifdef CONFIG_POSIX_API
#include <stropts.h>
#else
#include <zephyr/posix/stropts.h>
#endif
```

I see this pattern in headers tests.

Reply:
I think so, yeah

Reply:
Have I added them correctly? Does the if-block only need to be valid for methods?

Reply:
Yes - that's right

Reply:
Should I rebase and squash if everything looks right? Thanks!

Reply:
> Should I rebase and squash if everything looks right? Thanks!

Ideally yes